server/server/data_loader/change_tracker.py

```diff
@@ -0,0 +1,110 @@
+import hashlib
+import inspect
+import gc
+
+class ChangeTracker:
+    def __init__(self, base_dir, db):
+        self.base_dir = base_dir
+        self.db = db
+        
+        self.old_function_hashes = {doc['_key']: doc['hash'] for doc in db.aql.execute('''
+            FOR doc IN function_hashes
+                RETURN doc
+        ''')}
+        self.new_function_hashes = {}
+        
+        # Extract the mtimes from arangodb
+
+        self.old_mtimes = old_mtimes = {entry['path']: entry['mtime'] for entry in db.aql.execute('''
+        FOR entry in mtimes
+            RETURN entry
+        ''')}
+
+        # Get the mtimes from the file system
+
+        self.new_mtimes = new_mtimes = {}
```

Comment:
you should not assign the same object to 2 variables here. Just use self.new_mtimes everywhere